optuna/integration/_lightgbm_tuner/_train.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+from typing import Any
+from typing import Callable
```

Comment:
@c-bata If I remember correctly, you mentioned that we would like to somehow remove `abc` at some points, but does this also apply to this as well?
Just in case, I would to make sure.

Reply:
I think we should follow the PRP584. As in https://github.com/optuna/optuna/issues/4508, we change the import like this.

While I can't be certain and this is just a conjecture, I believe the recommendation to avoid the abc module likely refers to when utilizing polymorphism. Previously in Optuna, polymorphism was used by defining abstract classes with abc.ABCMeta, but recently, we've been using polymorphism through duck typing using typing.Protocol. See https://peps.python.org/pep-0544/ for more details.